src/catalog/src/kvbackend/manager.rs

```diff
@@ -303,36 +301,70 @@ impl CatalogManager for KvBackendCatalogManager {
             }
         });
 
-        let table_id_stream = self
-            .table_metadata_manager
-            .table_name_manager()
-            .tables(catalog, schema)
-            .map_ok(|(_, v)| v.table_id());
         const BATCH_SIZE: usize = 128;
-        let user_tables = try_stream!({
+        const CONCURRENCY: usize = 8;
+
+        let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
+        let metadata_manager = self.table_metadata_manager.clone();
+        let catalog = catalog.to_string();
+        let schema = schema.to_string();
+        tokio::task::spawn(async move {
+            let table_id_stream = metadata_manager
+                .table_name_manager()
+                .tables(&catalog, &schema)
+                .map_ok(|(_, v)| v.table_id());
             // Split table ids into chunks
             let mut table_id_chunks = table_id_stream.ready_chunks(BATCH_SIZE);
 
+            let mut task_batch = Vec::with_capacity(CONCURRENCY);
             while let Some(table_ids) = table_id_chunks.next().await {
-                let table_ids = table_ids
+                let table_ids = match table_ids
                     .into_iter()
                     .collect::<std::result::Result<Vec<_>, _>>()
                     .map_err(BoxedError::new)
-                    .context(ListTablesSnafu { catalog, schema })?;
-
-                let table_info_values = self
-                    .table_metadata_manager
-                    .table_info_manager()
-                    .batch_get(&table_ids)
-                    .await
-                    .context(TableMetadataManagerSnafu)?;
-
-                for table_info_value in table_info_values.into_values() {
-                    yield build_table(table_info_value)?;
+                    .context(ListTablesSnafu {
+                        catalog: &catalog,
+                        schema: &schema,
+                    }) {
+                    Ok(table_ids) => table_ids,
+                    Err(e) => {
+                        let _ = tx.send(Err(e));
+                        return;
+                    }
+                };
+
+                let metadata_manager = metadata_manager.clone();
+                let tx = tx.clone();
+                task_batch.push(tokio::spawn(async move {
```

Comment:
Why not use one of our common runtimes to spawn the tasks? It makes the async tasks better governed and monitored in the future.

Reply:
> Why not use one of our common runtimes to spawn the tasks? It makes the async tasks better governed and monitored in the future.

fixed in [24ff86b](https://github.com/GreptimeTeam/greptimedb/pull/4737/commits/24ff86bfb5809f4ef6ac5b56ac2c0f727762f62e)